src/components/CardList/CardList.jsx

```diff
@@ -1,17 +1,44 @@
 import "./CardList.css";
-import CardContext from "../../store";
-import { useContext } from "react";
+import { v4 as uuidv4 } from "uuid";
+import { useSelector, useDispatch } from "react-redux";
+import { cardActions } from "../../store/card-slice";
+import axios from "axios";
 
 import Card from "./../UI/Card";
+let isInitial = true;
 
 const CardList = ({ isViewMode }) => {
-  const ctx = useContext(CardContext);
+  const dispatch = useDispatch();
 
-  return ctx.errorMessage ? (
-    <div className="error-message">{ctx.errorMessage}</div>
+  const cardsData = useSelector((state) => state.card.cardsText);
+  const errorMessage = useSelector((state) => state.card.errorMessage);
+
+  if (isInitial) {
+    axios
+      .get(
+        "https://raw.githubusercontent.com/BrunnerLivio/PokemonDataGraber/master/output.json"
+      )
+      .then((response) => {
+        const respData = response.data.slice(0, 15).map((data) => {
+          return {
+            id: uuidv4(),
+            headerText: data.Name,
+            bodyText: data.About,
+          };
+        });
+        dispatch(cardActions.getInitialCards({ initialCards: respData }));
+      })
+      .catch((error) => {
+        dispatch(cardActions.setErrorMessage({ errorMessage: error.message }));
+      });
+    isInitial = false;
+  }
```

Comment:
At this point you are making CardList un-reusable. Data load should be better performed at app initialization, not specific component. Also better move this load to redux.